core/src/components/alert/alert-interface.ts

```diff
@@ -45,7 +45,7 @@ type AlertButtonOverlayHandler = boolean | void | { [key: string]: any };
 
 export interface AlertButton {
   text: string;
-  role?: 'cancel' | 'destructive' | string;
+  role?: 'cancel' | 'destructive' | Omit<string, 'cancel' | 'destructive'>;
```

Comment:
We want to avoid having to re-type the role in multiple places.

Instead, I would consider something similar to what we did for colors to solve a similar problem when using a type union with `string`:

https://github.com/ionic-team/ionic-framework/pull/25347/files#diff-83006355bd18aced9f88fd3b27ff6a40527c713b314145c2a34ad54b6361fa87

Reply:
Is there a global interface similar to what I mentioned in the _Other information_ section of the initial comment?
If not, I will implement what you have suggested, and continue for now. However, I would suggest there to be something similar, as Ionic is flexible and these interfaces, I believe, are useful.

Reply:
👋 currently the `interface.d.ts` file acts as the global interface. `LiteralUnion` is currently an internal type, but we could export it to be used for these changes as well: https://github.com/ionic-team/ionic-framework/blob/main/core/src/interface.d.ts#L132

Reply:
Hi, I've exported the LiteralUnion type 407ee7c328fb52d67e84550c39eadf9ef7fa8e87, then updated the unions of `ActionSheet`, `Alert`, and `Toast` 67adfba4504c5e28b76f5be46ad196a4ad791761.

Reply:
Awesome, thank you! I'll review and add another member of the team for an additional review as well 👍